comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -108,13 +119,27 @@ func newRemoteConfigClient(deps dependencies) (rcclient.Component, error) {
 		clientHA:      clientHA,
 	}
 
+	deps.Lc.Append(fx.Hook{
+		OnStart: func(context.Context) error {
+			if config.IsRemoteConfigEnabled(config.Datadog) {
+				return rc.start()
+			}
+			return nil
+		},
+	})
+
+	deps.Lc.Append(fx.Hook{
+		OnStop: func(context.Context) error {
+			rc.client.Close()
+			return nil
+		},
+	})
+
 	return rc, nil
 }
 
 // Start subscribes to AGENT_CONFIG configurations and start the remote config client
-func (rc rcClient) Start(agentName string) error {
-	rc.client.SetAgentName(agentName)
-
+func (rc rcClient) start() error {
```

Comment:
Doesn't look like this can return an `error` anymore (which is good, because returning an `error` to `OnStart` prevents the entire fx lifecycle from starting up). I'd remove the return type just to make it even harder to return an error here in the future